Hakawai/Mentions/HKWMentionsPlugin.m

```diff
@@ -1606,6 +1606,16 @@ - (BOOL)textViewShouldChangeTextInRangeV2:(NSRange)range
     if ([text length] == 0 && range.length == 1) {
         return [self advanceStateForCharacterDeletionV2:range.location];
     }
+    if ([text length] == 1 && range.length == 0 && self.mentions.count > 0) {
+        [self.mentions enumerateObjectsUsingBlock:^(HKWMentionsAttribute *mention, NSUInteger idx __unused, BOOL *stop) {
```

Comment:
this is great, I just think attribute:atIndex:effectiveRange: would be faster here. so like 

NSRange range;
[attribute:HKWMentionsAttribute atIndex:range.location effectiveRange:&range];
[self bleachExistingMentionAtRange:range];

with appropriate if guarding

- once you do this, you can remove the self.mentions.count in the if

Reply:
Thanks @bkoatz for reviewing PR.

- We have tried to use `attribute(at:effectiveRange:)` instead of iterating over `mentions`.
- When `whitespace` is encountered/inserted at end. It's not considered in `self.parentTextView.attributedText` range.
- So `attribute(at:effectiveRange:)` throws error related to out of bound issue because it checks beyond text range.
- This is one of scenario, we have look for other such corner cases if exits.

Following code we have tried to implement :
```
NSRange mentionRange;
[self.parentTextView.attributedText attribute:HKWMentionAttributeName atIndex:range.location effectiveRange:&mentionRange];
if (mentionRange.location != NSNotFound) {
      [self bleachExistingMentionAtRange:mentionRange];
}
```

- We think that `attribute(at:effectiveRange:)` will be as fast as iterating over `mentions` .
- Because we stopping in both cases if we found out cursor lies within particular mention.
- And both are looping over array of elements, so complexity of both will be `O(n)`.
- So we think that iterating over mentions would be safer in this case.

Please let us know if we are missing something

Reply:
great catch with the whitespace!

the reason im worried about self.mentions is because it iterates over the whole string and does multiple comparisons and instantiates and creates objects - it may be same big(O) time (im not sure), but it's definitely more complex.

Let's get in this PR first. https://github.com/linkedin/Hakawai/pull/171 If it works with currentlySelectedMentionRangeV2, we can just use that, to see if the character was inserted within the currently selected mention range (because the range will always be selected if the cursor is within it).

If not: 1) Does doing range.location-1 work for the enumeration? if you do max(range.location -1, 0)? I think it should
2) if not, then we can keep this mention thing for now

Reply:
Used `currentlySelectedMentionRangeV2` to bleach mention.